### PR TITLE
Adds more tests for iteration analytics workflow functions

### PR DESCRIPTION
**Pull request by edubarrosTNO**
Contributes to #31

### CODE REVIEW

**Comment by edubarrosTNO** [comment]
In the process of adding more tests for the functions introduced for analytics workflow, I observed that `normalize_data` function had a small issue. This is now fixed

**Comment by edubarrosTNO** [comment]
Added some more tests: we are now at 55.48%

File: src/flownet/ahm/_ahm_iteration_analytics.py [hunk 1]
```diff
@@ -38,60 +38,57 @@ def filter_dataframe(df_in, key_filter, min_value, max_value):
...
-        df[keys].values,
-        (df[keys].values.shape[0] * df[keys].values.shape[1], 1),
-        order="F",
-    )
-    data = np.tile(data, n)
+    keys = df_opm.keys()
+    keys = keys[df_opm.keys().str.contains(str_key)]
+    data = df_opm[keys].values
+    data = data.flatten()
+    data = np.transpose(np.tile(data, (n_real, 1)))
```
Comment (anders-kiaer) [inline src/flownet/ahm/_ahm_iteration_analytics.py:63]: Have not tested, but could this be simplified to e.g.
```suggestion
    data = np.transpose(np.tile(df_opm[keys].values.flatten(), (n_real, 1)))
```

File: src/flownet/ahm/_ahm_iteration_analytics.py [hunk 2]
```diff
@@ -38,60 +38,57 @@ def filter_dataframe(df_in, key_filter, min_value, max_value):
...
         to one realization of the ensemble
     """
 
-    keys = df.keys()
-    keys = keys[df.keys().str.contains(str_key)]
-    data = df[keys].values
+    keys = df_flownet.keys()
+    keys = keys[df_flownet.keys().str.contains(str_key)]
+    data = df_flownet[keys].values
     data = data.flatten()
```
Comment (anders-kiaer) [inline src/flownet/ahm/_ahm_iteration_analytics.py:90]:
```suggestion
    data = df_flownet[keys].values.flatten()
```

File: src/flownet/ahm/_ahm_iteration_analytics.py [hunk 3]
```diff
@@ -38,60 +38,57 @@ def filter_dataframe(df_in, key_filter, min_value, max_value):
...
-    data = np.reshape(data, (int(data.shape[0] / n), n), order="F")
+    data = np.reshape(data, (int(data.shape[0] / n_real), n_real), order="F")
```
Comment (anders-kiaer) [inline src/flownet/ahm/_ahm_iteration_analytics.py:91]:
```suggestion
    data = np.reshape(data, (data.shape[0] // n_real, n_real), order="F")
```

File: src/flownet/ahm/_ahm_iteration_analytics.py [hunk 4]
```diff
@@ -296,15 +294,18 @@ def make_dataframe_simulation_data(path, eclbase_file, keys):
...
-        df_tmp["DATE"] = pd.Series(dates)
-        df_tmp["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
+            df_sim["DATE"] = pd.Series(realizations_dict[runpath].dates)
+            df_sim["REAL_ID"] = pd.Series(
+                id_real * np.ones(len(realizations_dict[runpath].dates)), dtype=int
+            )
+        df_tmp["DATE"] = pd.Series(realizations_dict[runpath].dates)
+        df_tmp["REAL_ID"] = pd.Series(
+            id_real * np.ones(len(realizations_dict[runpath].dates)), dtype=int
+        )
```
Comment (anders-kiaer) [inline src/flownet/ahm/_ahm_iteration_analytics.py:308]: What is the reasoning behind not creating the variable `dates` (which is used some times in this code block)?

File: tests/test_calculation_accuracy_metric.py [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
 def test_calculation_accuracy_metric() -> None:
     assert (
         accuracy_metric(
-            [[0.5, 0.5, 0.5], [0.5, 0.5, 0.5]],
-            [[0.4, 0.4, 0.4], [0.6, 0.6, 0.6]],
+            np.array([[0.5, 0.5, 0.5], [0.5, 0.5, 0.5]]),
+            np.array([[0.4, 0.4, 0.4], [0.6, 0.6, 0.6]]),
             "RMSE",
         )
         == 0.09999999999999998
```
Comment (anders-kiaer) [inline tests/test_calculation_accuracy_metric.py:13]: Maybe this test should use something like [`math.isclose`](https://docs.python.org/dev/library/math.html#math.isclose) in order to not get a false negative due to floating point issues?

File: src/flownet/ahm/_ahm_iteration_analytics.py (hunk 4 above)
Comment (edubarrosTNO) [reply src/flownet/ahm/_ahm_iteration_analytics.py:308]: `pylint` was complaining of "too many locals"

File: tests/test_calculation_accuracy_metric.py (hunk 5 above)
Comment (edubarrosTNO) [reply tests/test_calculation_accuracy_metric.py:13]: good point! I have never used this, but will take a look into it

File: src/flownet/ahm/_ahm_iteration_analytics.py (hunk 4 above)
Comment (anders-kiaer) [reply src/flownet/ahm/_ahm_iteration_analytics.py:308]: Ah, ok. Maybe we can keep it, and simply (at least for now) disable that pylint message in this function? 🙂 
http://pylint.pycqa.org/en/latest/faq.html#is-it-possible-to-locally-disable-a-particular-message

**Pull request merged**